selectors/views/threads.ts

```diff
@@ -72,3 +91,87 @@ export const isThreadOpen = (state: GlobalState, threadId: $ID<UserThread>): boo
 export const isThreadManuallyUnread = (state: GlobalState, threadId: $ID<UserThread>): boolean => {
     return state.views.threads.manuallyUnread[threadId] || false;
 };
+
+// Returns a selector that, given the state and an object containing an array of postIds and an optional
+// timestamp of when the channel was last read, returns a memoized array of postIds interspersed with
+// day indicators and an optional new message indicator.
+export function makeFilterRepliesAndAddSeparators() {
```

Comment:
How much of this is shared with `makeFilterPostsAndAddSeparators`? Could those two be combined?

Reply:
That was my initial intent but unfortunately the conditions to mark as unread don't match between the two, and furthermore in replies we have one condition more, I could though refactor so they both share common logic.

Reply:
If they end up being different enough, I think keeping them separate is fine. I'd rather not complicate things by forcing them together since we should realistically only ever have the two places they're used and that's it. Their code also presumably won't diverge much since I can't see us touching this very often once CRT is stabilized.